server/server.go

```diff
@@ -202,12 +203,36 @@ func NewServer(config *Config) (*Server, error) {
 		m.executor.GenesisPostHook = factory(m.config.Chain, engineName)
 	}
 
-	// compute the genesis root state
-	genesisRoot, err := m.executor.WriteGenesis(config.Chain.Genesis.Alloc)
+	var initialStateRoot = types.ZeroHash
+
+	if ConsensusType(engineName) == PolyBFTConsensus {
+		polyBFTConfig, err := consensusPolyBFT.GetPolyBFTConfig(config.Chain)
+		if err != nil {
+			return nil, err
+		}
+
+		if polyBFTConfig.InitialTrieRoot != types.ZeroHash {
+			checkedInitialTrieRoot, err := itrie.HashChecker(polyBFTConfig.InitialTrieRoot.Bytes(), stateStorage)
+			if err != nil {
+				return nil, fmt.Errorf("error on state root verification %w", err)
+			}
+
+			if checkedInitialTrieRoot != polyBFTConfig.InitialTrieRoot {
+				return nil, errors.New("invalid initial state root")
+			}
+
+			logger.Warn("Initial state root checked and correct")
```

Comment:
```suggestion
			logger.Info("Initial state root checked and correct")
```

Reply:
Should we continue executing if this condition is not satisfied?